taf/git.py

```diff
@@ -150,15 +150,32 @@ def remotes(self) -> List[str]:
         return self._remotes
 
     @property
-    def is_git_repository_root(self) -> bool:
-        """Check if path is git repository."""
-        git_path = self.path / ".git"
-        return self.is_git_repository and (git_path.is_dir() or git_path.is_file())
+    def is_git_repository(self) -> bool:
+        try:
+            repo = pygit2.Repository(str(self.path))
```

Comment:
It's better to use `self.pygit_repo` repo here because this property handles errors when instantiating pygit2 repository instances